bin/install-wp-cli-tests.sh

```diff
@@ -0,0 +1,222 @@
+	if [[ $WP_VERSION =~ [0-9]+\.[0-9]+(\.[0-9]+)? ]]; then
+		WP_TESTS_TAG="tags/$WP_VERSION"
+	else
+		# get the latest version of wordpress from core installed
+		LATEST_VERSION=$(wp core version)
+		if [[ -z "$LATEST_VERSION" ]]; then
+			echo "Latest WordPress version could not be found"
+			exit 1
+		fi
+		WP_TESTS_TAG="tags/$LATEST_VERSION"
+	fi
+}
+
+install_test_suite() {
+	get_installed_wp_version
+	# portable in-place argument for both GNU sed and Mac OSX sed
+	if [[ $(uname -s) == 'Darwin' ]]; then
+		local ioption='-i .bak'
+	else
+		local ioption='-i'
+	fi
+
+	# set up testing suite if it doesn't yet exist
+	if [ ! -d "$WP_TESTS_DIR" ]; then
+		# set up testing suite
+		mkdir -p $WP_TESTS_DIR
+		svn co --quiet https://develop.svn.wordpress.org/"${WP_TESTS_TAG}"/tests/phpunit/includes/ "${WP_TESTS_DIR}/includes"
+	fi
+
+	cd "${WP_TESTS_DIR}"
+
+	if [ ! -f wp-tests-config.php ]; then
+		download https://develop.svn.wordpress.org/${WP_TESTS_TAG}/wp-tests-config-sample.php "${WP_TESTS_DIR}/wp-tests-config.php"
+		sed $ioption "s:dirname( __FILE__ ) . '/src/':'$WP_CORE_DIR':" "$WP_TESTS_DIR"/wp-tests-config.php
+		sed $ioption "s/youremptytestdbnamehere/$DB_NAME/" "$WP_TESTS_DIR"/wp-tests-config.php
+		sed $ioption "s/yourusernamehere/$DB_USER/" "$WP_TESTS_DIR"/wp-tests-config.php
+		sed $ioption "s/yourpasswordhere/$DB_PASS/" "$WP_TESTS_DIR"/wp-tests-config.php
+		sed $ioption "s|localhost|${DB_HOST}|" "$WP_TESTS_DIR"/wp-tests-config.php
+	fi
+}
+
+create_plugins_folder(){
+	if [ ! -d "${WP_CORE_DIR}wp-content/plugins" ]; then
+		cd "${WP_CORE_DIR}wp-content/"
+		mkdir plugins
+	fi
+}
+
+install_plugins(){
+	create_plugins_folder
+	# Create symlink to current path
+	ln -s "${INITIAL_WORKING_DIRECTORY}" "${CURRENT_PLUGIN}"
+	# After running install_test_suite need to return to directory $WP_CORE_DIR where wp-cli installed
+	cd "${WP_CORE_DIR}"
+	# Install plugins
+	wp plugin activate "$WP_PLUGINS"
+}
+
+install_themes(){
+	# Install the theme
+	wp theme install "$WP_THEMES" --activate
+}
+
+delete_db(){
+	# Deletes the existing database.
+	wp db drop --yes
+}
+
+delete_the_installed_directories(){
+  # Delete the $WP_CORE_DIR directory and all its contents, including any subdirectories and files
+  if [ -d "$WP_CORE_DIR" ]; then
+    rm -rf "$WP_CORE_DIR"
+    if [ ! -d "$WP_CORE_DIR" ]; then
+      echo "Directory $WP_CORE_DIR is deleted."
+    else
+		  echo "Directory $WP_CORE_DIR does not delete."
+		fi
+	fi
+
+	# Delete the $WP_TESTS_DIR directory and all its contents, including any subdirectories and files
+	if [ -d "$WP_TESTS_DIR" ]; then
+    rm -rf "$WP_TESTS_DIR"
+    if [ ! -d "$WP_TESTS_DIR" ]; then
+      echo "Directory $WP_TESTS_DIR is deleted."
+    else
+		  echo "Directory $WP_TESTS_DIR does not delete."
+		fi
+	fi
+}
+
+delete_symlink(){
+	# Check if file exists and is a symbolic link (same as -h command)
+	if [ -L "${CURRENT_PLUGIN}" ]; then
+		unlink "${CURRENT_PLUGIN}"
+		echo "deleted symlink ${CURRENT_PLUGIN}"
```

Comment:
All debug/log data should be printed via function that print with a format like 
`Date [info|warning|error] - Message.`

Reply:
Fixed I added print_msg function as you wish

Reply:
You should identify each message type (`info|warning|error`)